### ISSUE TITLE
Stock Candles no_data

### ISSUE THREAD

**Issue by TestingDebugging** [finnhubio/Finnhub-API#313]
## What is the issue ? Plz describe in details

Stock candle calls with the same parameters sometimes returns no_data.

## How to reproduce the error ?

Call
 https://finnhub.io/api/v1/stock/candle?symbol=AAPL&resolution=1&from=1572651390&to=1572910590&token=[API Key]

multiple times and it will randomly fail sometimes and return {"s":"no_data"}.

You even seem to have this Issue in your own widget in the API documentation. When you refresh the page multiple times the widget won't have any data sometimes.

![Screenshot_20200710-171717~2.png](https://user-images.githubusercontent.com/64195695/87170943-4f71cc00-c2d2-11ea-9f77-e83fe4da8c84.png)

## What is the expected result and why do you think it's the correct result ?

Expected:  Candlestick Data

**Comment by finnhubio** [comment]
Hi,

We are experiencing exponential daily growth in the number of users and requests. Because of our generous free plan, we are working hard to keep up with the demands. This issue with candles data should be resolved this week.

Our apology for the inconvenience :)

**Comment by finnhubio** [comment]
We have scheduled to increase the number of cpus in our datacenter. Feel free to reach out again if you still encounter any difficulty next week

**Comment by TestingDebugging** [comment]
Thank you for the quick response much appreciated!
Will let you know if I encounter any difficulty next week.

**Comment by Z223I** [comment]
I too am getting erratic  responses to my candle stick requests.  If I switch to a paid account, would there be consistent results?

**Comment by finnhubio** [comment]
> I too am getting erratic responses to my candle stick requests. If I switch to a paid account, would there be consistent results?

Yes you should receive better performance with a paid account as premium users' requests get the priority queue.

**Comment by barocsi** [comment]
@finnhubio This issue still persists. Please confirm that paid users get proper and reliable responses comparing to the free tier, also please include this in your plans description so we can quickly decide to move on with this service or to go find one that is appropriate for production environments.

**Comment by barocsi** [comment]
Update: after subscribing to the service, I still get no_data errors in a random fashion; Customer support is silent.

**Comment by igrabes** [comment]
@finnhubio Any update here, we are seeing the same issue?

**Comment by stephenrs** [comment]
> Update: after subscribing to the service, I still get no_data errors in a random fashion; Customer support is silent.

@barocsi @igrabes Unfortunately, you'll learn that either these guys are in over their heads with regard to the technology required to reliably deliver the data they promise to deliver, or they are just not interested in doing it. It's hard to tell which, but this service is simply not production caliber. I've been waiting for resolutions to similar issues for about 2 months, while additional new issues appear for me almost daily. Customer service has gone completely silent, not responding to my emails, so I've stopped wasting my time reporting them.

I've been hopeful that finnhub would be able to stabilize their infrastructure, but that hasn't happened, so my hope has diminished accordingly.

**Comment by barocsi** [comment]
So far I have only got one response from support asking me to provide more examples of the issue.

**Comment by stephenrs** [comment]
@barocsi Yes, it's a frustrating experience. I began testing the API in May 2020, and reported this particular issue back then along with several other issues that fit into the same category. As you noted above, this problem occurs randomly, and there are a few other similar issues that exhibit the same random API behavior. These are chronic problems, and I've reported most of the issues that I've observed, along with clear steps to reproduce them, but so far none of the issues have been addressed.

What's particularly surprising and disappointing about all of this is that this category of issues affects all API users (including their own widget, as has been pointed out) and are very easy to reproduce, so one would think that they would be high priority, but unfortunately as the months have gone by it seems that this is not the case. Additionally, the lack of transparency and honesty about these chronic issues is disturbing. Bugs are a fact of life, but making non-credible excuses or pretending that they don't exist is not acceptable.

Also, as I write this, the finnhub.io domain is completely inaccessible...again. This is the third time I've seen this happen in the last 2-3 months. Each time it has happened, the API and all other Finnhub web properties have been offline for 4-6 hours...with no credible explanation as to why or what is being done to prevent it from happening again.

Finnhub shows some promise, but the infrastructure unfortunately has some quite major and very obvious flaws.

**Comment by GeorgeLeePatterson** [comment]
Finnhub python library consistently returns no_data while, intermittently, the url requests do. I am fine not using the python API to access your data even if you put it on your site as the library to use, but when I go to access the data via the URL it is completely hit or miss. Is anyone going to respond to this thread? Are refunds in order?? (I am a paid subscriber, seems I should have put my money to better use)

**Comment by barocsi** [comment]
Thanks for the feedback it clearly resonates with my very short term experience and indicates how careful should one be when considering options. I have received a response about the outage it was a prompt statement that the outage is fixed without any further indication, not even addressing the randomly occurring data errors.

**Comment by stephenrs** [comment]
Yes, again, it's unfortunate. They are happy to take your money, but getting a straight and professional answer from them about anything is impossible...despite how cheerful and helpful they pretend to be in public forums.
